snippets/card-product.liquid

```diff
@@ -336,21 +336,35 @@
     </div>
   </div>
 {%- else -%}
+  {%- liquid
+    assign ratio = 1
+    if media_aspect_ratio == 'portrait'
+      assign ratio = 0.8
+    elsif media_aspect_ratio == 'adapt'
+      assign ratio = card_product.featured_media.aspect_ratio
+    endif
+    if ratio == 0 or ratio == null
+      assign ratio = 1
+    endif
+  -%}
   <div class="card-wrapper product-card-wrapper underline-links-hover">
     <div
       class="
         card card--{{ settings.card_style }}
         {% if extend_height %} card--extend-height{% endif %}
+        {% if image_shape and image_shape != 'default' %} card--shape{% endif %}
         {% if settings.card_style == 'card' %} color-{{ settings.card_color_scheme }} gradient{% endif %}
       "
-      style="--ratio-percent: 100%;"
+      style="--ratio-percent: {{ 1 | divided_by: ratio | times: 100 }}%;"
     >
       <div
         class="card__inner{% if settings.card_style == 'standard' %} color-{{ settings.card_color_scheme }} gradient{% endif %} ratio"
-        style="--ratio-percent: 100%;"
+        style="--ratio-percent: {{ 1 | divided_by: ratio | times: 100 }}%;"
       >
-        <div class="card__media">
-          <div class="media media--transparent">
+        <div class="card__media {% if image_shape and image_shape != 'default' %} shape--{{ image_shape }} color-{{ settings.card_color_scheme }} gradient{% endif %}"
+        style="{% if  image_shape == 'default' and media_aspect_ratio == 'portrait' %} top: 3rem;{% endif %}">
+          <div class="media media--transparent"
+            style="{% if  image_shape != 'default' and media_aspect_ratio == 'portrait' %} top: 3rem;{% endif %}">
```

Comment:
We can also remove the option to change the media ratio since this is not a common practice with placeholder and the scope of this PR applies to image shapes :)

Reply:
I went ahead and removed the option to change the media ratio in the next commit since it's not a common practice with placeholder and the scope of this PR applies to image shapes :)

Reply:
Coming back around to this PR after some time away. 🙂 I think removing the aspect ratio is a little weird since it means that the aspect ratio control has no effect. The goal of this PR is to make sure one of our settings has a visible effect on the preview state, so it seems weird to make another setting have no effect while we're at it. 

But I agree it's less weird than having the background color showing through. Do we have any other possible solutions in mind for this?

Reply:
Ah yes — I think if we have no clear solution for this yet, it's ok to defer the image ratio portion of the work.